iosApp/iosApp/ContentView.swift

```diff
@@ -346,6 +357,15 @@ struct ContentView: View {
         }
     }
 
+    private func recenterOnVehicleButtonInfo() -> (RouteType, Vehicle, Stop)? {
+        guard case let .stopDetails(stopId: _, stopFilter: stopFilter, tripFilter: _) = nearbyVM
+            .navigationStack.lastSafe(),
+            let selectedVehicle = mapVM.selectedVehicle, let globalData = mapVM.globalData,
+            let stop = nearbyVM.getTargetStop(global: globalData), let routeId = stopFilter?.routeId,
+            let route = globalData.routes[routeId] else { return nil }
```

Comment:
This doesn't work for GL because its filter "routeId" is actually the line ID. I think you could use the one from the vehicle instead?

Reply:
I thought about using the vehicle's route ID but decided not to because the vehicle's route may not actually be the currently selected route. This was silly, though, because buses will not spontaneously become trains. However, if the vehicle does not have a route assigned at all, it'd be nice if this button kept working; I'm not sure what conditions would cause that to happen, but there appear to currently be zero vehicles with no route, so it should be rare enough that we can get away with falling back to the stop filter route ID.

Reply:
Yeah I wonder if it's possible for the vehicles to have a null route when coming from a garage or being in between routes. I ran into basically the same issue with the trip details card and created [TripRouteAccents](https://github.com/mbta/mobile_app/blob/main/iosApp/iosApp/ViewModels/StopDetailsViewModel.swift#L23-L41) for just the general display related components of a route, maybe it's worth splitting that out into its own file and using it here too, though it seems much less risky in this case that someone will get confused and attempt to use the route for something that requires that it be the correct trip route.